src/js/components/sharedComponents/UploadFile.tsx

```diff
@@ -89,13 +97,57 @@ const UploadFile = (): JSX.Element => {
         body: formData
       })
         .then(response => response.json())
-        .catch(e => console.log('error in onDropFile()', e));
+        .catch(e => console.log('fetching error in onDropFile()', e));
+
+      const arcGISResults = geojsonToArcGIS(featureCollection.data.attributes);
+
+      await Promise.all(
+        arcGISResults.map(async (feature: FeatureResult) => {
+          const registeredGeometry = await registerGeometry(feature)
+            .then((response: Response) =>
+              response.status === 200 ? response.json() : null
+            )
+            .catch((e: Error) => {
+              // TODO [ ] - error handling logic (to account for when one geometry produces an error)
+              console.log(
+                'error using registerGeometry() in UploadFile.tsx',
+                e
+              );
+            });
+          feature.attributes.geostoreId = registeredGeometry.data.id;
+
+          return registeredGeometry;
+        })
+      )
+        .then((registeredGeometries: any) => {
+          if (registeredGeometries.length) {
+            const oldActiveFeatures = [...activeFeatures];
+            const graphics = mapController.generateGraphics(arcGISResults);
+
+            const shapeFileFeatures: LayerFeatureResult = {
+              layerID: 'upload_file_features',
+              layerTitle: 'Upload File Features',
+              sublayerID: null,
+              sublayerTitle: null,
+              features: graphics
+            };
+
+            oldActiveFeatures.push(shapeFileFeatures);
+            dispatch(setActiveFeatures(oldActiveFeatures));
```

Comment:
This logic is leaving artifacts of previous upload. In the data panel if you upload a second file after the first, they both show up and it has some regressions, we need to clean out the activeFeature array with only single item (our shapeFileFeatures). Also from feature result we do not need anything else except attributes and geometry. Check out suggestion below.

```suggestion
            const graphics = mapController.generateGraphics(arcGISResults);

            const shapeFileFeatures: LayerFeatureResult = {
              layerID: 'upload_file_features',
              layerTitle: 'Upload File Features',
              sublayerID: null,
              sublayerTitle: null,
              features: graphics.map((g: __esri.Graphic) => {
                return { attributes: g.attributes, geometry: g.geometry };
              })
            };

            dispatch(setActiveFeatures([shapeFileFeatures]));
```